### PR TITLE
User post CRUD.

### PR DESCRIPTION
**Pull request by gopalbbharadva**
#### Features:
- Users can create, update and delete posts. (image also)
- Users can see their own posts on its profile page.

P.S: for now feed page will not show posts of users' following accounts. I will do it  later.

live link:https://deploy-preview-9--sm-spotter.netlify.app/

### CODE REVIEW

**Review by mimansha-swarup — APPROVED** [review]
code well written
meaningful names
and clean code

File: src/components/Postmodal/PostModal.jsx [hunk 1]
```diff
@@ -38,15 +78,68 @@ export const PostModal = () => {
...
+        >
+          {image === "" && postData?.postImage === "" ? (
+            <img className="h-20" src={DummyImage} />
+          ) : (
+            <img
+              src={`${
+                postData?.postImage !== "" && image === ""
+                  ? postData?.postImage
+                  : image
+              }`}
```
Comment (msdeshmukh009) [inline src/components/Postmodal/PostModal.jsx]: Try to give `alt` attributes for these images

File: src/features/postSlice.js [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+);
+
+export const postUserPost = createAsyncThunk(
+  "post/postUserPost",
+  async ({ postData, token }, { rejectWithValue }) => {
+    try {
+      const res = await postService(postData, token);
+      if (res.status === 201) return res.data;
+    } catch (error) {
+      console.log(error);
```
Comment (msdeshmukh009) [inline src/features/postSlice.js]: You can remove this console

File: src/page/Posts/components/PostFeatureModal/PostFeatureModal.jsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          }}
+        >
+          <VscChromeClose />
+        </button>
+        <button
+          onClick={() => {
+            dispatch(postEdit(true));
+            dispatch(hidePostFeatureModal());
+            dispatch(showPostModal());
+          }}
```
Comment (msdeshmukh009) [inline src/page/Posts/components/PostFeatureModal/PostFeatureModal.jsx]: It's better to extract this function out of your jsx

**Review by msdeshmukh009 — APPROVED** [review]
The CRUD operations are working as expected. Nice work!!

**Review by pritam-kr — APPROVED** [review]

**Pull request merged**